### PR TITLE
Refactor stream and consumer not created checks in Eventing FI tests

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added tests for stream and consumer not created checks based on eventing-sink function.

- Test run for `pre-main-kyma-gardener-gcp-subscription-upgrade` - [sample](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/16942/mfaizanse_test_of_prowjob_pre-main-kyma-gardener-gcp-subscripti/1630495794467442688)

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/16840

### CODE REVIEW

File: tests/fast-integration/eventing-test/eventing-test.js [hunk 1]
```diff
@@ -322,6 +327,56 @@ describe('Eventing tests', function() {
     });
   }
 
+  function jsSaveStreamAndConsumersDataSuite() {
+    context('save stream and consumer data', function() {
+      it('saving JetStream data to test stream and consumers will not be re-created by kyma upgrade', async function() {
+        if (!isEventingSinkDeployed || !testSubscriptionV1Alpha2) {
```
Comment (k15r) [inline tests/fast-integration/eventing-test/eventing-test.js:333]: why only with V1Alpha2?

Comment (mfaizanse) [reply tests/fast-integration/eventing-test/eventing-test.js:333]: v1alpha1 is currently covered in [here](https://github.com/kyma-project/kyma/blob/dbbe74c5ee167a0e9561dd23bc5e0b7d08d20d28/tests/fast-integration/eventing-test/eventing-test.js#L327), which is based of the subscription created for last-order function. Later, when we have the new release, we can cleanup all this code.

**Review by k15r — APPROVED** [review]

**Pull request merged**